### PR TITLE
Remove experimental flag for custom client directives

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

- Remove `experimental.customClientDirectives` config (and its docs)

RFC: https://github.com/withastro/roadmap/pull/583

## Testing

n/a. existing tests should pass.

## Docs

Remove experimental reference: https://github.com/withastro/docs/pull/3376

### CODE REVIEW

**Review by sarah11918 — APPROVED** [review]
Bye bye, `customClientDirectives: true`, we hardly knew you!  :smiling_face_with_tear:

**Comment by natemoo-re** [comment]
Code looks great other than failing `astro check`, unrelated?

**Comment by bluwy** [comment]
Yeah, the astro check tests has been flaky recently

File: packages/astro/src/core/config/schema.ts [hunk 1]
```diff
@@ -43,8 +43,6 @@ const ASTRO_CONFIG_DEFAULTS: AstroUserConfig & any = {
 	experimental: {
 		assets: false,
 		hybridOutput: false,
-		customClientDirectives: false,
-		middleware: false,
```
Comment (bluwy) [inline packages/astro/src/core/config/schema.ts:47]: Looks like this wasn't removed in the unflag middleware PR, which I removed here.

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**